src/middleware/secure-headers/index.test.ts

```diff
@@ -189,284 +194,289 @@ describe('Secure Headers Middleware', () => {
     const app = new Hono()
-    app.use(
-      '/test',
-      secureHeaders({
-        contentSecurityPolicy: {
-          defaultSrc: ["'self'"],
-          baseUri: ["'self'"],
-          fontSrc: ["'self'", 'https:', 'data:'],
-          frameAncestors: ["'self'"],
-          imgSrc: ["'self'", 'data:'],
-          objectSrc: ["'none'"],
-          scriptSrc: ["'self'"],
-          scriptSrcAttr: ["'none'"],
-          styleSrc: ["'self'", 'https:', "'unsafe-inline'"],
-        },
-      })
-    )
 
-    app.all('*', async (c) => {
-      c.res.headers.set('Strict-Transport-Security', 'Hono')
-      return c.text('header updated')
+    app.get('/test', secureHeaders(), poweredBy(), async (c) => {
+      return c.text('Hono is cool')
     })
 
-    const res = await app.request('/test')
-    expect(res.headers.get('Content-Security-Policy')).toEqual(
-      "default-src 'self'; base-uri 'self'; font-src 'self' https: data:; frame-ancestors 'self'; img-src 'self' data:; object-src 'none'; script-src 'self'; script-src-attr 'none'; style-src 'self' https: 'unsafe-inline'"
-    )
-  })
-
-  it('CSP Setting one only', async () => {
-    const app = new Hono()
-    app.use(
-      '/test',
+    app.get(
+      '/test2',
       secureHeaders({
-        contentSecurityPolicy: {
-          defaultSrc: ["'self'"],
-        },
-      })
+        removePoweredBy: false,
+      }),
+      poweredBy(),
+      async (c) => {
+        return c.text('Hono is cool')
+      }
     )
 
-    app.all('*', async (c) => {
-      return c.text('header updated')
-    })
-
     const res = await app.request('/test')
-    expect(res.headers.get('Content-Security-Policy')).toEqual("default-src 'self'")
-  })
-
-  it('No CSP Setting', async () => {
-    const app = new Hono()
-    app.use('/test', secureHeaders({ contentSecurityPolicy: {} }))
-
-    app.all('*', async (c) => {
-      return c.text('header updated')
-    })
+    const poweredby = res.headers.get('X-Powered-By')
+    expect(poweredby).toEqual(null)
+    expect(await res.text()).toEqual('Hono is cool')
 
-    const res = await app.request('/test')
-    expect(res.headers.get('Content-Security-Policy')).toEqual('')
+    const res2 = await app.request('/test2')
+    const poweredby2 = res2.headers.get('X-Powered-By')
+    expect(poweredby2).toEqual('Hono')
+    expect(await res2.text()).toEqual('Hono is cool')
   })
 
-  it('CSP with reportTo', async () => {
-    const app = new Hono()
-    app.use(
-      '/test1',
-      secureHeaders({
-        reportingEndpoints: [
-          {
-            name: 'endpoint-1',
-            url: 'https://example.com/reports',
+  describe.each([
+    { cspSettingName: 'contentSecurityPolicy', cspHeaderName: 'Content-Security-Policy' },
+    {
+      cspSettingName: 'contentSecurityPolicyReportOnly',
+      cspHeaderName: 'Content-Security-Policy-Report-Only',
+    },
+  ])('CSP Setting ($cspSettingName)', ({ cspSettingName, cspHeaderName }) => {
```

Comment:
Diff is crazy... but the following just adds this loop and moves the X-powered-by test to the top.